src/components/Button/Button.tsx

```diff
@@ -60,30 +63,39 @@ const Button = forwardRef<HTMLButtonElement | HTMLAnchorElement, ButtonProps>(
       isDisabled = false,
       href,
       target,
+      prefetch = 'none',
       ...props
     },
     ref,
   ) => {
     const { className: baseClassName, ...commonProps } = getCommonProps(props, 'Button');
+    const [linkHovered, setLinkHovered] = useState(false);
     if (href) {
       return (
-        <a
-          {...commonProps}
-          ref={ref as React.ForwardedRef<HTMLAnchorElement>}
-          href={href}
-          className={classnames(
-            `${baseClassName}`,
-            `${baseClassName}--${variant}`,
-            {
-              [`${baseClassName}--icon-last`]: iconLast,
-            },
-            className,
-          )}
-          target={target}
-          rel={target === '_blank' ? 'noopener noreferrer' : undefined}
-        >
-          {children}
-        </a>
+        <>
+          <a
+            {...commonProps}
+            ref={ref as React.ForwardedRef<HTMLAnchorElement>}
+            href={href}
+            className={classnames(
+              `${baseClassName}`,
+              `${baseClassName}--${variant}`,
+              {
+                [`${baseClassName}--icon-last`]: iconLast,
+              },
+              className,
+            )}
+            target={target}
+            rel={target === '_blank' ? 'noopener noreferrer' : undefined}
+            onMouseOver={() => setLinkHovered(true)}
+            onMouseLeave={() => setLinkHovered(false)}
+            onClick={props.onClick}
+          >
+            {children}
+          </a>
+          {prefetch === 'intent' && linkHovered && <link data-testid="prefetch-link" rel="prefetch" href={href} />}
```

Comment:
can't this be a `prefetch === 'intent' || prefetch === 'render'`?

Reply:
will mousing over and mousing out on repeat cause this link to be removed over and over again from the DOM.  Maybe we shouldn't turn off the linkHovered on mouseLeave since the prefetch will have already started?

Reply:
yea, we could just leave it in there if we want